src/util/metrics/threads_linux.rs

```diff
@@ -158,48 +152,31 @@ impl Collector for ThreadsCollector {
 
 /// Gets thread ids of the given process id.
 pub fn get_thread_ids(pid: pid_t) -> Result<Vec<pid_t>> {
-    let mut tids = Vec::new();
-    let dirs = fs::read_dir(format!("/proc/{}/task", pid))?;
-    for task in dirs {
-        let file_name = match task {
-            Ok(t) => t.file_name(),
-            Err(e) => {
-                error!("fail to read task of {}, error {:?}", pid, e);
-                continue;
-            }
-        };
-
-        let tid = match file_name.to_str() {
-            Some(tid) => match tid.parse() {
-                Ok(tid) => tid,
+    Ok(fs::read_dir(format!("/proc/{}/task", pid))?
+        .filter_map(|task| {
+            let file_name = match task {
+                Ok(t) => t.file_name(),
                 Err(e) => {
                     error!("fail to read task of {}, error {:?}", pid, e);
-                    continue;
+                    return None;
+                }
+            };
+
+            match file_name.to_str() {
+                Some(tid) => match tid.parse() {
+                    Ok(tid) => Some(tid),
+                    Err(e) => {
+                        error!("fail to read task of {}, error {:?}", pid, e);
```

Comment:
Please refine these logs to follow the Unified Log Format later.

Reply:
Hope I'm doing it correctly.